src/domain/prices/prices.repository.ts

```diff
@@ -16,21 +16,29 @@ export class PricesRepository implements IPricesRepository {
     nativeCoinId: string;
     fiatCode: string;
   }): Promise<number> {
-    const result = await this.coingeckoApi.getNativeCoinPrice(args);
+    const lowerCaseFiatCode = args.fiatCode.toLowerCase();
+    const result = await this.coingeckoApi.getNativeCoinPrice({
+      nativeCoinId: args.nativeCoinId,
+      fiatCode: lowerCaseFiatCode,
+    });
     const assetPrice = await this.assetPriceValidator.validate(result);
-    const { nativeCoinId, fiatCode } = args;
-    return assetPrice?.[nativeCoinId]?.[fiatCode];
+    return assetPrice?.[args.nativeCoinId]?.[lowerCaseFiatCode];
   }
 
   async getTokenPrice(args: {
     chainName: string;
     tokenAddress: string;
     fiatCode: string;
   }): Promise<number> {
-    const result = await this.coingeckoApi.getTokenPrice(args);
+    const lowerCaseFiatCode = args.fiatCode.toLowerCase();
+    const lowerCaseTokenAddress = args.tokenAddress.toLowerCase();
```

Comment:
Why are we converting the token address to lower case?

Reply:
For context: Coingecko converts the token address to lowercase internally, so even if we send a _randomcase_ address, we need to check for a lowercase key in the response. This was already taken into consideration before this change:

```
return assetPrice?.[tokenAddress.toLowerCase()]?.[fiatCode];
```

So having said that it's fair to point out that the lowercase conversion before calling `CoingeckoApi` is not needed, but for consistency, given the API will return lowercase addresses anyway, I took the decision to send them in lowercase also. This also covers us in the (improbable) case that Coingecko starts refusing non-lowercase addresses.

But not strong opinion on this, if you think it's better to just call `.toLowerCase()` when checking the response I'm also OK with it.

Reply:
No strong opinion on this either. I would leave a comment though on why we change the `tokenAddress` to lower case.

Also the `PricesRepository` is in the domain and we are tying an implementation detail of Coingecko to the domain instead of the `IPricesApi` (Coingecko specific). Making it lowercase, and setting/transforming the respective keys seems to be more appropriate.

Reply:
> Also the `PricesRepository` is in the domain and we are tying an implementation detail of Coingecko to the domain instead of the `IPricesApi` (Coingecko specific). Making it lowercase, and setting/transforming the respective keys seems to be more appropriate.

I totally agree with you on this. In fact, I had thought about it as well while writing this PR. This is more a technical design issue that was present before this PR, as the lowercase parsing was already in place for the `PricesApi` response. I also think this repository contains logic that should be moved to the Coingecko-specific classes, so I'm going to address that in a separate PR and merge this to address the bug it solves as soon as possible. Thanks!